### PR TITLE
first batch of small changes

### PR DESCRIPTION
**Pull request by andreistefanescu**

### CODE REVIEW

File: semantics/language/common/symloc.k [hunk 1]
```diff
@@ -42,9 +42,10 @@ module C-SYMLOC-SYNTAX
 
 
      // Base, byte offset, bit offset.
-     syntax SymLoc ::= loc(SymBase, Int, Set)
      syntax SymLoc ::= loc(SymBase, Int)
-     syntax SymLoc ::= "NullPointer"
+     syntax SymLoc ::= HeapSymLoc
```
Comment (ellisonch) [inline semantics/language/common/symloc.k]: There is no heap in C :)

Comment (dwightguth) [inline semantics/language/common/symloc.k]: I'd go with CanonicalSymLoc

File: semantics/language/common/typing/misc.k [hunk 2]
```diff
@@ -331,6 +331,7 @@ module C-TYPING-MISC
      rule getReadFrom'(K:K, S:Set) => getReadFrom'(.K, S)
           requires getKLabel(K) =/=KLabel 'readFrom
                andBool (K =/=K .K)
+     rule getReadFrom'(.K, .Set) => NullPointer
```
Comment (dwightguth) [inline semantics/language/common/typing/misc.k]: this does not seem correct to me... I believe the motivation is supposed to be that this function is only called in contexts where hasReadFrom has already evaluated to true. Maybe we should create some kind of special bottom value if you need the function to be total, but NullPointer is a valid SymLoc, which is not where this particular value was actually read from.

File: semantics/language/common/symloc.k (hunk 1 above)
Comment (andreistefanescu) [inline semantics/language/common/symloc.k]: Sounds good.

**Comment by andreistefanescu** [comment]
Jenkins: test this please

**Comment by andreistefanescu** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by andreistefanescu** [comment]
@dwightguth please review

**Pull request merged**